### DISCUSSION TITLE
how to get the curl.exe in the build.cake file ?

### DISCUSSION THREAD

**Discussion by ElangoRajendran** [cake-build/cake#3292] [Q&A] [answered]
I need to post the comments on the GitHub Pull request page using Curl. Also, the action should be performed in the build.cake file. I have a used below curl command for posting the comment on the pull request page,
This is the command,

**curl -H "Authorization: Token 6XXXXXXXa" -X POST -d "{ \"body\": \"My comments \" }" "https://api.github.com/repos/username/reponame/issues/pullrequestnumber/comments**"

My question is how we can get the curl.exe in build.cake file and how to use this command in build.cake file?
Could you suggest a solution for this?

**Comment by AdmiringWorm** [comment]
One approach to make use of curl inside a cake build script would be to use the [`Cake.Curl`](https://www.nuget.org/packages/Cake.Curl/) addin. This would of course require you to have curl installed on the system that is running the script.

A different approach would be to [`StartProcess`](https://cakebuild.net/dsl/process/) helpers to manually launch the curl.exe program, and pass in the arguments that you need.

**Comment by augustoproiete** [comment, marked as answer]
@ElangoRajendran Out of curiosity, why do you need to use Curl specifically to perform this task?

One of the main advantages of Cake is to be able to use C# and easily leverage existing NuGet packages, so you could use [Octokit](https://github.com/octokit/octokit.net), GitHub's official API Client library for .NET, to interact with GitHub's API without having to manually craft URLs and JSON payloads, etc.

```cakescript
#addin nuget:?package=Octokit&version=0.50.0

using Octokit;

var github = new GitHubClient(new ProductHeaderValue("ElangoRajendran"))
{
    Credentials = new Credentials(token: "6XXXXXXXa"),
};

var pullRequestNumber = 123;
var commentBody = "My comments";

github.Issue.Comment.Create("username", "reponame", pullRequestNumber, commentBody)
    .GetAwaiter().GetResult();
```

**Reply by ElangoRajendran** [reply]
Hi @augustoproiete,
Thanks for the update,
while trying to run the above code I have faced the below issue,
![image](https://user-images.githubusercontent.com/59197465/112802955-6d65a380-9090-11eb-864e-497d2643e6a8.png)
Then, checked my token, its got deleted. Finally, I have confirmed while try to run the above code the token got deleted automatically.
Could you suggest a solution to this issue?

**Reply by augustoproiete** [reply]
@ElangoRajendran I suspect that you are committing the token to your GitHub repo, GitHub will remove the token automatically if that's the case.

**Tokens should be a parameter to your Cake build script, or an environment variable**, etc.

If you're not committing the token to your GitHub repo, my next guess is that you/your company have a webhook set up in the GitHub user/organization to delete tokens after use.

**Reply by ElangoRajendran** [reply]
Hi @augustoproiete ,
I am not using the webhook, also we are not deleting the token.

This is my code,
		
		var github = new GitHubClient(new ProductHeaderValue("ElangoRajendran"))
		{
		    Credentials = new Credentials(token: Mytoken),
		};

		var pullRequestNumber = 15;
		var commentBody = "My comments-update";

		github.Issue.Comment.Create("ElangoRajendran", "my-repo-name", pullRequestNumber, commentBody)
		    .GetAwaiter().GetResult();

Mytoken value has passed as a parameter.

While running the above code token got deleted automatically from my account and got error occurred. But run the same code in PowerShell directly it's working fine. I have to face this issue only running the code GitHub workflow actions with build.cake file.

Could suggest a solution for this issue?

**Reply by augustoproiete** [reply]
Where is `Mytoken` coming from?

**Reply by ElangoRajendran** [reply]
Hi @augustoproiete ,

Mytoken value has come from Github actions workflow. Passed parameter as PowerShell script to build.cake file

**Reply by augustoproiete** [reply]
Do you have the token written in plaintext anywhere in any file that gets saved in the repository?

If you're using GitHub Actions, you should be using the Secrets feature to get the API key - https://docs.github.com/en/actions/reference/encrypted-secrets

Are you?

**Reply by ElangoRajendran** [reply]
Hi @augustoproiete ,

Thanks for the update, it's working fine.

**Reply by augustoproiete** [reply]
Great to hear @ElangoRajendran ! What was it in the end?
